src/components/TextInput/native/TextInput.js

```diff
@@ -0,0 +1,233 @@
+    }
+    if (isFocused) {
+      this.setState({ isFocused: false });
+    }
+  };
+
+  render() {
+    const {
+      name,
+      label,
+      showPlaceholder,
+      error: errorMessage,
+      style,
+      theme,
+      leftIcon,
+      rightIcon,
+      showRightIconWhenFocusedOnly,
+      testID,
+      accessibilityLabel,
+      ...props
+    } = this.props;
+    const { formik } = this.context;
+    const { labelTranslateValue, isFocused } = this.state;
+
+    const inputProps = { name };
+    let error = errorMessage;
+    const labelFontSize = labelTranslateValue.interpolate({
+      inputRange: [0, theme.pxScale * 0.8],
+      outputRange: [parseInt(theme.fontSize.xxs, 10), parseInt(theme.fontSize.s, 10)],
+      extrapolate: 'clamp',
+    });
+
+    inputProps.onChangeText = this.handleTextChange;
+    inputProps.onBlur = this.handleBlur;
+    inputProps.onFocus = this.onFocus;
+    if (formik) {
+      inputProps.value = formik.values[name];
+      error = formik.touched[name] && formik.errors[name];
+    }
+
+    return (
+      <InputContainer style={style}>
+        <BorderedContainer
+          error={error}
+          isFocused={isFocused}
+        >
+          {
+            leftIcon
+          }
+          <RelativeFlexView>
+            {
+              !showPlaceholder ? (
+                <AnimatedLabel
+                  error={error}
+                  isFocused={isFocused}
+                  isEmptyInput={!inputProps.value}
+                  style={{
+                    fontSize: labelFontSize,
+                    transform: [
+                      { translateY: labelTranslateValue },
+                      { perspective: 1000 },
+                    ],
+                  }}
+                >
+                  {label}
+                </AnimatedLabel>
+              ) : null
+            }
+            <Input
+              id={name}
+              placeholder={showPlaceholder ? label : ''}
+              underlineColorAndroid={theme.color.transparent}
+              selectionColor={theme.color.green}
+              isLabelVisible={!showPlaceholder}
+              testID={testID}
+              accessibilityLabel={accessibilityLabel}
+              {...inputProps}
+              {...props}
+            />
+          </RelativeFlexView>
+          {
+            !showRightIconWhenFocusedOnly || isFocused ? rightIcon : null
+          }
+        </BorderedContainer>
+        {
+          error ? (
+            <ErrorText>
+              {error}
+            </ErrorText>
+          ) : null
+        }
+      </InputContainer>
+    );
+  }
+}
+
+TextInput.propTypes = {
+  ...Input.propTypes,
+  name: PropTypes.string,
+  label: PropTypes.string,
+  showPlaceholder: PropTypes.bool,
+  error: PropTypes.string,
+  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+  theme: PropTypes.object,
+  leftIcon: PropTypes.node,
+  rightIcon: PropTypes.node,
+  showRightIconWhenFocusedOnly: PropTypes.bool,
```

Comment:
How about combining the logic for `showRightIconWhenFocusedOnly` and `rightIcon` with
```
rightIcon={</> || {
  component: </>
  showOnlyOnFocus: false
}}
```